Build warnings in samples/net/cloud/google_iot_mqtt

When building this sample I get:

NOTE: to build this sample you need before cmake:
```
cd samples/net/cloud/google_iot_mqtt/src/private_info
./create_keys.py -d foobar
```

```
[ 28%] Built target drivers__entropy
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c: In function 'show_addrinfo':
/home/galak/git/zephyr/include/logging/log_core.h:229:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
  229 |   printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
      |          ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:257:5: note: in expansion of macro 'Z_LOG_TO_PRINTK'
  257 |     Z_LOG_TO_PRINTK(_level, __VA_ARGS__);        \
      |     ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:2: note: in expansion of macro 'LOG_DBG'
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:27: note: format string is defined here
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/include/logging/log_core.h:146:24: warning: format '%lx' expects argument of type 'long unsigned int', but argument 4 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
  146 | #define Z_LOG_STR(...) "%s: " GET_ARG1(__VA_ARGS__), __func__\
      |                        ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:179:30: note: in definition of macro '__LOG_INTERNAL'
  179 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:270:14: note: in expansion of macro 'Z_LOG_STR'
  270 |              Z_LOG_STR(__VA_ARGS__));\
      |              ^~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:2: note: in expansion of macro 'LOG_DBG'
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:27: note: format string is defined here
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |          ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                          |
      |                                                          uint32_t {aka unsigned int}
/home/galak/git/zephyr/include/logging/log_core.h:179:30: note: in definition of macro '__LOG_INTERNAL'
  179 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:2: note: in expansion of macro 'LOG_DBG'
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:27: note: format string is defined here
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 2 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |          ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                          |
      |                                                          uint32_t {aka unsigned int}
/home/galak/git/zephyr/include/logging/log_core.h:282:27: note: in definition of macro '__LOG'
  282 |    log_printf_arg_checker(__VA_ARGS__);         \
      |                           ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:2: note: in expansion of macro 'LOG_DBG'
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:83:27: note: format string is defined here
   83 |  LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/include/logging/log_core.h:229:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
  229 |   printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
      |          ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:257:5: note: in expansion of macro 'Z_LOG_TO_PRINTK'
  257 |     Z_LOG_TO_PRINTK(_level, __VA_ARGS__);        \
      |     ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:2: note: in expansion of macro 'LOG_DBG'
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:27: note: format string is defined here
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/include/logging/log_core.h:146:24: warning: format '%lx' expects argument of type 'long unsigned int', but argument 4 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
  146 | #define Z_LOG_STR(...) "%s: " GET_ARG1(__VA_ARGS__), __func__\
      |                        ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:179:30: note: in definition of macro '__LOG_INTERNAL'
  179 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:270:14: note: in expansion of macro 'Z_LOG_STR'
  270 |              Z_LOG_STR(__VA_ARGS__));\
      |              ^~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:2: note: in expansion of macro 'LOG_DBG'
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:27: note: format string is defined here
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |          ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                          |
      |                                                          uint32_t {aka unsigned int}
/home/galak/git/zephyr/include/logging/log_core.h:179:30: note: in definition of macro '__LOG_INTERNAL'
  179 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:287:2: note: in expansion of macro '__LOG'
  287 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:2: note: in expansion of macro 'LOG_DBG'
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:27: note: format string is defined here
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:9:
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:10: warning: format '%lx' expects argument of type 'long unsigned int', but argument 2 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |          ^~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                          |
      |                                                          uint32_t {aka unsigned int}
/home/galak/git/zephyr/include/logging/log_core.h:282:27: note: in definition of macro '__LOG'
  282 |    log_printf_arg_checker(__VA_ARGS__);         \
      |                           ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:2: note: in expansion of macro 'LOG_DBG'
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |  ^~~~~~~
/home/galak/git/zephyr/samples/net/cloud/google_iot_mqtt/src/main.c:84:27: note: format string is defined here
   84 |  LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
      |                         ~~^
      |                           |
      |                           long unsigned int
      |                         %x
[ 28%] Linking C static library libsubsys__net__lib__dns.a
```

--- a/samples/net/cloud/google_iot_mqtt/src/main.c
+++ b/samples/net/cloud/google_iot_mqtt/src/main.c
@@ -6,10 +6,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <logging/log.h>
-
-LOG_MODULE_REGISTER(net_google_iot_mqtt, LOG_LEVEL_INF);
-
 #include <zephyr.h>
 
 #include "dhcp.h"
@@ -18,6 +14,11 @@ LOG_MODULE_REGISTER(net_google_iot_mqtt, LOG_LEVEL_INF);
 #include <net/sntp.h>
 #include <net/net_config.h>
 #include <net/net_event.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(net_google_iot_mqtt, LOG_LEVEL_INF);
 
 /* This comes from newlib. */
 #include <time.h>
@@ -80,8 +81,8 @@ top:
 	LOG_DBG("  addrlen : %d", (int)addr->ai_addrlen);
 
 	/* Assume two words. */
-	LOG_DBG("   addr[0]: 0x%lx", ((uint32_t *)addr->ai_addr)[0]);
-	LOG_DBG("   addr[1]: 0x%lx", ((uint32_t *)addr->ai_addr)[1]);
+	LOG_DBG("   addr[0]: 0x%x", ((uint32_t *)addr->ai_addr)[0]);
+	LOG_DBG("   addr[1]: 0x%x", ((uint32_t *)addr->ai_addr)[1]);
 
 	if (addr->ai_next != 0) {
 		addr = addr->ai_next;

